app/policies/request_policy.rb

```diff
@@ -29,6 +29,15 @@ def complete?
     true
   end
 
+  def update?
+    if record.request_distributions.exists?
+      record.errors.add(:base, 'cannot update a request with associated request distributions')
+      false
+    else
+      true
+    end
+  end
```

Comment:
ok pour la logique 👌 si on veut l'écrire de manière un peu plus "rails-ish" et concise, ça pourrait donner quelque chose comme ça :
```suggestion
  def update?
    return true unless record.request_distributions.exists?

    record.errors.add(:base, 'cannot update a request with associated request distributions')
    false
  end
```